plugins/kubernetes-actions/src/actions/createKubernetesNamespace.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  ActionContext,
+  createTemplateAction,
+} from '@backstage/plugin-scaffolder-node';
+import {
+  KubeConfig,
+  CoreV1Api,
+  HttpError,
+  V1Namespace,
+} from '@kubernetes/client-node';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Entity } from '@backstage/catalog-model';
+
+const KUBERNETES_API_URL_ANNOTATION = 'kubernetes.io/api-server';
+const KUBERNETES_CLUSTER_TYPE = 'kubernetes-cluster';
+
+export interface HttpErrorBody {
+  kind: string;
+  apiVersion: string;
+  metadata: Object;
+  status: string;
+  message: string;
+  reason: string;
+  details: { name: string; kind: string };
+  code: number;
+}
+
+type TemplateActionParameters = {
+  namespaceName: string;
+  clusterRef?: string;
+  url?: string;
+  token: string;
+  skipTLSVerify?: boolean;
+  caData?: string;
+};
+
+const getUrlFromClusterRef = async (
+  ctx: ActionContext<TemplateActionParameters>,
+  catalogClient: CatalogClient,
+  clusterRef: string,
+): Promise<string> => {
+  const isResource = clusterRef.startsWith('resource:');
+  if (!isResource) {
+    ctx.logger.warn(
+      'Cluster reference in the wrong format, attempting to fix it',
+    );
+  }
+  const catalogEntity: Entity | undefined = await catalogClient.getEntityByRef(
+    isResource ? clusterRef : `resource:${clusterRef}`,
+  );
+  if (!catalogEntity) {
+    throw new Error('Resource not found');
+  }
+  if (catalogEntity.spec?.type !== KUBERNETES_CLUSTER_TYPE) {
+    ctx.logger.warn(`Resource is not of ${KUBERNETES_CLUSTER_TYPE} type`);
+  }
+  const apiUrl =
+    catalogEntity.metadata?.annotations?.[KUBERNETES_API_URL_ANNOTATION];
+  if (!apiUrl) {
+    throw new Error(
+      `Cluster resource is missing ${KUBERNETES_API_URL_ANNOTATION} annotation`,
+    );
+  }
+  return apiUrl;
+};
+
+const validateUrl = (url: string | undefined) => {
+  try {
+    // eslint-disable-next-line no-new
+    new URL(url!);
+  } catch (error) {
+    throw new Error(`"${url}" is an invalid url`);
+  }
+};
+
+export function createKubernetesNamespaceAction(catalogClient: CatalogClient) {
+  return createTemplateAction<TemplateActionParameters>({
+    id: 'kubernetes:create-namespace',
+    description: 'Creates a kubernetes namespace',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['namespaceName', 'token'],
```

Comment:
Should also contain `oneOf`:
https://json-schema.org/understanding-json-schema/reference/combining.html?highlight=oneof#oneof

```
oneOf: [{required: ['url']}, {required: ['clusterRef']} ]
```